packages/api/schema/mutation.js

```diff
@@ -880,6 +880,20 @@ export default [
       Deletes the specified payment credential.
       """
       removePaymentCredentials(paymentCredentialsId: ID!): PaymentCredentials
+      """
+      Sign a generic payment provider for registration
+      """
+      signPaymentProviderForCredentialRegistration(
+        paymentProviderId: ID!
+      ): String
+
+      """
+      Sign a generic order payment
+      """
+      signPaymentProviderForCheckout(
```

Comment:
Maybe we should use a non-optional String as return type. If a payment provider does not support signing we could throw that as error like "This Payment Provider does not support signing transactions" or something like that.

Reply:
I think we can safely make it return a non-optional String because it's already making that check inside the resolver and `OrderPayment.sign` is called in GENERIC type only so...